core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/factory/OnboardingInstitutionStrategyFactory.java

```diff
@@ -63,10 +61,14 @@ public OnboardingInstitutionStrategy retrieveOnboardingInstitutionStrategy(Insti
             digestOnboardingInstitutionStrategy = ignore -> {};
             persitOnboardingInstitutionStrategy = verifyManagerAndPersistWithDigest();
             emailsOnboardingInstitutionStrategy = ignore -> {};
-        } else {
+        } else if (InstitutionType.PA == institutionType || Boolean.TRUE.equals(checkIfGspProdInteropAndOriginIPA(institutionType, productId, institution.getOrigin()))) {
```

Comment:
you could avoid `Boolean.TRUE.equals(` check.... `checkIfGspProdInteropAndOriginIPA` method returns a boolean